src/main/Boards/AdjMatrix.java

```diff
@@ -0,0 +1,90 @@
+//This class is borrowed directly from the Trevorah Hex Game Project.
+//It allows for quick calculations of hex tile adjacency. Before it can
+//function properly, the Matrix Toolkit library (mtj-1.0.8.jar) must be 
+//added to the project as a dependency.
+
+package main.Boards;
+
+import java.util.ArrayList;
+import no.uib.cipr.matrix.Matrix;
+import no.uib.cipr.matrix.UpperSPDPackMatrix;
+
+public class AdjMatrix {
+	public static final int NO_LINK = 0;
+	  public static final int LINK = 1;
+	  private Matrix data;
+	  private int size;
+
+	  public AdjMatrix(int size) {
+	    this.size = size;
+	    this.data = new UpperSPDPackMatrix(size);
+	  }
+
+	  public AdjMatrix(Matrix adj) {
+	    this.data = adj;
+	    this.size = data.numColumns();
+	  }
+
+	  @Override
+	  public AdjMatrix clone() {
+	    return new AdjMatrix(data.copy());
+	  }
+
+	  public Matrix getData() {
+	    return this.data;
+	  }
+
+	  public AdjMatrix mult(AdjMatrix m2) {
+	    if (m2.size() == this.size()) {
+	      Matrix retMatrix = new UpperSPDPackMatrix(size);
+	      data.mult(m2.getData(), retMatrix);
+	      return new AdjMatrix(retMatrix);
+	    } else
+	      return null;
+	  }
+
+	  public void write(int a, int b, int value) {
+	    if (a == b)
+	      value = LINK;
+	    internalWrite(a, b, value);
+	  }
+
+	  public int read(int x, int y) {
+	    return (int) data.get(x, y);
+	  }
```

Comment:
For type safety, we should use Integer.tryParse(data.get(x, y)) and wrap it in a try catch. In the catch we should throw the exception up with a message about where it came from.